In [5]:
import os

os.chdir("C:\\Learning\\Lets-Upgrade\\Day 8\\Assignment")

In [3]:
import pandas as pd

dataset1 = pd.read_excel('Attrition_general_data.xlsx',sheetname=0)

d = {'No': 0, 'Yes': 1}
dataset1['Attrition'] = dataset1['Attrition'].map(d).fillna(dataset1['Attrition'])
dataset1['Attrition'].round(0).astype(int)
dataset1

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


Age  Attrition     BusinessTravel              Department  \
0      51          0      Travel_Rarely                   Sales   
1      31          1  Travel_Frequently  Research & Development   
2      32          0  Travel_Frequently  Research & Development   
3      38          0         Non-Travel  Research & Development   
4      32          0      Travel_Rarely  Research & Development   
5      46          0      Travel_Rarely  Research & Development   
6      28          1      Travel_Rarely  Research & Development   
7      29          0      Travel_Rarely  Research & Development   
8      31          0      Travel_Rarely  Research & Development   
9      25          0         Non-Travel  Research & Development   
10     45          0      Travel_Rarely  Research & Development   
11     36          0      Travel_Rarely  Research & Development   
12     55          0      Travel_Rarely  Research & Development   
13     47          1         Non-Travel  Research & Development   
14     28          0      Travel_Rarely  Research & Development   
15     37          0      Travel_Rarely  Research & Development   
16     21          0      Travel_Rarely  Research & Development   
17     37          0         Non-Travel  Research & Development   
18     35          0      Travel_Rarely                   Sales   
19     38          0      Travel_Rarely  Research & Development   
20     26          0  Travel_Frequently  Research & Development   
21     50          0      Travel_Rarely                   Sales   
22     53          0      Travel_Rarely  Research & Development   
23     42          0      Travel_Rarely  Research & Development   
24     29          0  Travel_Frequently  Research & Development   
25     55          0      Travel_Rarely  Research & Development   
26     26          0  Travel_Frequently  Research & Development   
27     37          0      Travel_Rarely                   Sales   
28     44          1  Travel_Frequently  Research & Development   
29     38          0      Travel_Rarely                   Sales   
...   ...        ...                ...                     ...   
4380   35          0  Travel_Frequently  Research & Development   
4381   29          1      Travel_Rarely  Research & Development   
4382   52          0      Travel_Rarely  Research & Development   
4383   42          0      Travel_Rarely  Research & Development   
4384   59          0      Travel_Rarely  Research & Development   
4385   50          0      Travel_Rarely  Research & Development   
4386   33          1      Travel_Rarely                   Sales   
4387   43          0      Travel_Rarely                   Sales   
4388   33          1      Travel_Rarely                   Sales   
4389   52          0         Non-Travel  Research & Development   
4390   32          0      Travel_Rarely         Human Resources   
4391   32          1      Travel_Rarely                   Sales   
4392   39          0      Travel_Rarely                   Sales   
4393   32          0         Non-Travel                   Sales   
4394   41          0      Travel_Rarely                   Sales   
4395   40          0      Travel_Rarely  Research & Development   
4396   45          0      Travel_Rarely  Research & Development   
4397   31          0  Travel_Frequently  Research & Development   
4398   33          0      Travel_Rarely  Research & Development   
4399   34          0      Travel_Rarely  Research & Development   
4400   37          0      Travel_Rarely  Research & Development   
4401   45          0  Travel_Frequently                   Sales   
4402   37          1  Travel_Frequently                   Sales   
4403   39          0  Travel_Frequently  Research & Development   
4404   29          0      Travel_Rarely                   Sales   
4405   42          0      Travel_Rarely  Research & Development   
4406   29          0      Travel_Rarely  Research & Development   
4407   25          0      Travel_Rarely  Research & D

In [7]:
dataset1.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [11]:
'''Appling Man Witney test
H0 = There is no significant difference between Age for both Attrition_yes and Attrition_no
H1 = There is significant difference between Age for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.Age, dataset_no.Age)

print(stats,p)

# Conclusion - Since p = 2.9951588479067175e-30 Probability value is < 0.05 which mean we can say that There is significant difference 
#between Age for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


961731.0 2.9951588479067175e-30


## Inferance of the above test with Attrition_yes and No vs Age
<io>
    <li>H0 = There is no significant difference between Age for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between Age for both Attrition_yes and Attrition_no</li>
</io>

### Here if we see that the probability value is 2.9951e-30 which is lesser than 0.05
### Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between Age for both Attrition_yes and Attrition_no

In [6]:
'''Appling Man Witney test
H0 = There is no significant difference between distanceFromHome for both Attrition_yes and Attrition_no
H1 = There is significant difference between distanceFromHome for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.DistanceFromHome, dataset_no.DistanceFromHome)

print(stats,p)

# Conclusion - Since p = 0.469 Probability value is > 0.05 which mean we can say that There is no significant difference between 
#DistanceFromHome for both Attrition_yes and Attrition_no (Null hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


1312110.0 0.4629185205822659


### Inferance of the above test with Attrition_yes and No vs DistanceFromHome
<io>
    <li>H0 = There is no significant difference between DistanceFromHome for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between DistanceFromHome for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 0.4629 which is greater than 0.05. 

### Therefore we can conclude that Null hypothesis is accepted i,e There is no significant difference between DistanceFromHome for both Attrition_yes and Attrition_no

In [7]:
'''Appling Man Witney test
H0 = There is no significant difference between PercentSalaryHike for both Attrition_yes and Attrition_no
H1 = There is significant difference between PercentSalaryHike for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.PercentSalaryHike, dataset_no.PercentSalaryHike)

print(stats,p)

# Conclusion - Since p = 0.0186601 Probability value is < 0.05 which mean we can say that There is significant difference between 
#PercentageSalaryHike for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


1250640.0 0.018660129917539733


### Inferance of the above test with Attrition_yes and No vs PercentSalaryHike
<io>
    <li>H0 = There is no significant difference between PercentSalaryHike for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between PercentSalaryHike for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 0.018660129 which is lesser than 0.05. 

###  Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between PercentSalaryHike for both Attrition_yes and Attrition_no

In [8]:
'''Appling Man Witney test
H0 = There is no significant difference between YearsSinceLastPromotion for both Attrition_yes and Attrition_no
H1 = There is significant difference between YearsSinceLastPromotion for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.YearsSinceLastPromotion, dataset_no.YearsSinceLastPromotion)

print(stats,p)

# Conclusion - Since p = 0.0002021 Probability value is < 0.05 which mean we can say that There is significant difference between 
#YearsSinceLastPromotion for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


1209366.0 0.0002021180346719736


### Inferance of the above test with Attrition_yes and No vs YearsSinceLastPromotion
<io>
    <li>H0 = There is no significant difference between YearsSinceLastPromotion for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between YearsSinceLastPromotion for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 0.0002021180346719736 which is lesser than 0.05. 

###  Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between YearsSinceLastPromotion for both Attrition_yes and Attrition_no

In [9]:
'''Appling Man Witney test
H0 = There is no significant difference between YearsWithCurrManager for both Attrition_yes and Attrition_no
H1 = There is significant difference between YearsWithCurrManager for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.YearsWithCurrManager, dataset_no.YearsWithCurrManager)

print(stats,p)

# Conclusion - Since p = 1.2365483142169853e-31 Probability value is < 0.05 which mean we can say that There is significant difference 
#between YearsWithCurrManager for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


957253.5 1.2365483142169853e-31


### Inferance of the above test with Attrition_yes and No vs YearsWithCurrManager
<io>
    <li>H0 = There is no significant difference between YearsWithCurrManager for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between YearsWithCurrManager for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 1.2365483142169853e-31 which is lesser than 0.05. 

### Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between YearsWithCurrManager for both Attrition_yes and Attrition_no

In [10]:
'''Appling Man Witney test
H0 = There is no significant difference between TotalWorkingYears for both Attrition_yes and Attrition_no
H1 = There is significant difference between TotalWorkingYears for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.TotalWorkingYears, dataset_no.TotalWorkingYears)

print(stats,p)

# Conclusion - Since p = 2.8897554531400447e-39 Probability value is < 0.05 which mean we can say that There is significant difference 
#between TotalWorkingYears for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


909954.0 2.8897554531400447e-39


### Inferance of the above test with Attrition_yes and No vs TotalWorkingYears
<io>
    <li>H0 = There is no significant difference between TotalWorkingYears for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between TotalWorkingYears for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 2.8897554531400447e-39 which is lesser than 0.05. 

### Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between TotalWorkingYears for both Attrition_yes and Attrition_no

In [12]:
'''Appling Man Witney test
H0 = There is no significant difference between YearsAtCompany for both Attrition_yes and Attrition_no
H1 = There is significant difference between YearsAtCompany for both Attrition_yes and Attrition_no

'''
import pandas as pd

dataset_yes= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_yes")
dataset_no= pd.read_excel('Attrition_general_data.xlsx',sheetname="Attrition_No")

from scipy.stats import mannwhitneyu

stats,p = mannwhitneyu(dataset_yes.YearsAtCompany, dataset_no.YearsAtCompany)

print(stats,p)

# Conclusion - Since p = 6.047598261692858e-37 Probability value is < 0.05 which mean we can say that There is significant difference 
#between TotalWorkingYears for both Attrition_yes and Attrition_no (Alternative Hypothesis is accepted)

C:\Users\rajusuja.CORPDOM\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)


923238.0 6.047598261692858e-37


### Inferance of the above test with Attrition_yes and No vs YearsAtCompany
<io>
    <li>H0 = There is no significant difference between YearsAtCompany for both Attrition_yes and Attrition_no</li>
    <li>H1 = There is significant difference between YearsAtCompany for both Attrition_yes and Attrition_no</li>
</io>

#### Here if we see that the probability value is 6.047598261692858e-37 which is lesser than 0.05. 

### Therefore we can conclude that Alternative hypothesis is accepted i,e There is significant difference between YearsAtCompany for both Attrition_yes and Attrition_no